napari/_qt/dialogs/qt_package_installer.py

```diff
@@ -201,6 +201,9 @@ def environment(
     ) -> QProcessEnvironment:
         if env is None:
             env = QProcessEnvironment.systemEnvironment()
+            # Fix for macOS when napari launched from terminal
+            # related to https://github.com/napari/napari/pull/5531
+            env.remove("PYTHONEXECUTABLE")
```

Comment:
Is env a reference or a copy? I think it should only be removed if it's a copy. I guess you can test this with:

```
env = QProcessEnvironment.systemEnvironment()
env.remove('PYTHONEXECUTABLE')
'PYTHONEXECUTABLE' in QProcessEnvironment.systemEnvironment()
```

If it's False, then I suggest:

```python
env = QProcessEnvironment.systemEnvironment()
```

Based on my reading of this, though, I think it's a copy anyway:

https://doc.qt.io/qtforpython/PySide6/QtCore/QProcessEnvironment.html#PySide6.QtCore.PySide6.QtCore.QProcessEnvironment.systemEnvironment

> This function does not cache the system environment. [...] note that repeated calls to this function will recreate the [QProcessEnvironment](https://doc.qt.io/qtforpython/PySide6/QtCore/QProcessEnvironment.html#PySide6.QtCore.PySide6.QtCore.QProcessEnvironment) object, which is a non-trivial operation.

Reply:
Your test code fails because `TypeError: argument of type 'QProcessEnvironment' is not iterable`
However:
```
from qtpy.QtCore import QProcess, QProcessEnvironment

env = QProcessEnvironment.systemEnvironment()

env.value('PYTHONEXECUTABLE')
Out[3]: '/Users/piotrsobolewski/Dev/miniforge3/envs/napari-main/bin/python3.10'

env.remove('PYTHONEXECUTABLE')

env2 = QProcessEnvironment.systemEnvironment()

env2.value('PYTHONEXECUTABLE')
Out[7]: '/Users/piotrsobolewski/Dev/miniforge3/envs/napari-main/bin/python3.10'
```
So it seems fine.
I'm not sure what the suggestion is, something missing?
<img width="580" alt="image" src="https://user-images.githubusercontent.com/76622105/224537993-f4a0532f-1994-4a3b-bb80-9aee9e126c94.png">

Reply:
I found another spot in the docs where it's very clear:

> The `QProcessEnvironment` class wraps that concept and allows easy manipulation of those variables. It’s meant to be used along with `QProcess` , to set the environment for child processes. It cannot be used to change the current process’s environment.

Reply:
Cool thanks!
So what do y'all think of putting this behind a set of `if` like the existing code here:
https://github.com/napari/napari/blob/9e911040148d232b76d768ed74ff90ac0fa6f707/napari/_qt/dialogs/qt_package_installer.py#L207-L215

Reply:
Hm yeah. I think PYTHONEXECUTABLE only affects macOS anyway but I think putting it in a block down with the other platform-specific code (i.e. that "nt" block) would make the motivation more clear and self-documenting.

Reply:
No, putting it in `if env is None` was just mindlessness and inexperience on my part.

I'll make a dedicated `if darwin` block for it.